src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -4595,6 +4595,45 @@ class spell_gen_impatient_mind : public SpellScriptLoader
         }
 };
 
+enum AuraReflectiveShields
+{
+    SPELL_KING_GUARD_REFLECTIVE_SHIELD_AURA      = 119630,
+    SPELL_KING_GUARD_REFLECTIVE_SHIELD_DAMAGE    = 119632
+};
+
+// 119630 - Reflective Shield.
+class spell_gen_kings_guard_reflective_shield : public SpellScriptLoader
+{
+    public:
+        spell_gen_kings_guard_reflective_shield() : SpellScriptLoader("spell_gen_kings_guard_reflective_shield") { }
+
+        class spell_gen_kings_guard_reflective_shield_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_gen_kings_guard_reflective_shield_AuraScript);
+
+            void OnProc(const AuraEffect* aurEff, ProcEventInfo& eventInfo)
+            {
+                Unit* caster = GetCaster();
+                Unit* target = GetTarget();
+                
+                if (eventInfo.GetDamageInfo()->GetAttacker() == target)
+                    return;
+                int32 reflectiveshielddmg = int32(CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), aurEff->GetAmount()));
+                caster->CastCustomSpell(SPELL_KING_GUARD_REFLECTIVE_SHIELD_DAMAGE, SPELLVALUE_BASE_POINT0, reflectiveshielddmg, eventInfo.GetDamageInfo()->GetAttacker(), true, NULL, aurEff);
```

Comment:
use nullptr instead NULL
And SPELL_KING_GUARD_REFLECTIVE_SHIELD_DAMAGE not need be validated?